pootle/core/url_helpers.py

```diff
@@ -12,7 +12,83 @@
 import urllib
 import urlparse
 
-from django.core.urlresolvers import reverse
+from django.core.urlresolvers import reverse, resolve
+from django.utils.functional import cached_property
+
+
+class PathResolver(object):
+    """Converts a request.path into parts for matching pootle_path
+    with a regex
+    """
+
+    def __init__(self, path):
+        self._path = path
+
+    @cached_property
+    def path(self):
+        if not self._path:
+            return self._path
+        parts = self._path.split("/")
+        if "." in parts[-1]:
+            return "/%s" % self._path.strip("/")
+        return "/%s/" % self._path.strip("/")
+
+    @cached_property
+    def kwa(self):
+        return resolve(self.path).kwargs
+
+    @property
+    def directory(self):
+        if not self.path:
+            return ""
+        directory = self.kwa.get("dir_path", "") or ""
+        if directory and not directory.endswith("/"):
+            directory = "%s/" % directory
+        return directory
+
+    @property
+    def filename(self):
+        if not self.path:
+            return ""
+        return self.kwa.get("filename", "") or ""
+
+    @property
+    def language(self):
+        if not self.path:
+            return None
+        return self.kwa.get("language_code", None) or None
+
+    @property
+    def project(self):
+        if not self.path:
+            return None
+        return self.kwa.get("project_code", None) or None
+
+    @property
+    def regex(self):
```

Comment:
Nit: Instead of adding the regex capability on the commit where `PathResolver` is added, I would add it on the "Use PathResolver.regex when filtering by pootle_path" commit. This way it is easier to see why some feature has been initially added.